hr_employee_calendar_planning/models/hr_employee.py

```diff
@@ -97,18 +97,54 @@ def _regenerate_calendar(self):
                         + " %s" % self.name,
                         "attendance_ids": vals_list,
                         "two_weeks_calendar": two_weeks,
+                        "tz": self.tz,  # take employee timezone as default
                     }
                 )
                 .id
             )
         else:
             self.resource_calendar_id.attendance_ids = vals_list
+
         # Set the hours per day to the last (top date end) calendar line to apply
         if self.calendar_ids:
             self.resource_calendar_id.hours_per_day = self.calendar_ids[
                 0
             ].calendar_id.hours_per_day
 
+            # set global leaves
+            self.resource_calendar_id.global_leave_ids = [
+                (
+                    6,
+                    0,
+                    self.copy_global_leaves(
+                        self._get_global_leave_ids(self.calendar_ids)
+                    ),
+                )
+            ]
+
+    def copy_global_leaves(self, leave_ids):
+        vals = [
+            leave.copy_data({})[0]
+            for leave in self.env["resource.calendar.leaves"].browse(leave_ids)
+        ]
+        return self.env["resource.calendar.leaves"].create(vals).ids
+
+    @staticmethod
```

Comment:
Don't add this kind of Python stuff that is not inheritabe in the Odoo ORM framework.